MyAwesomeSprint23

CapitanLatinamerica opened the pull request:
Если всё, то пора открывать бутылку 😁

CapitanLatinamerica committed:
SPRINT 23 часть 1
Сделал новую иконку

CapitanLatinamerica committed:
SPRINT 23 часть 1.1
Удалил из res все лишние картинки

CapitanLatinamerica committed:
SPRINT 23 часть 2
Создал новый фрагмент просмотра плейлиста

CapitanLatinamerica committed:
SPRINT 23 часть 3
Переход на просмотр плейлиста не раьотает

CapitanLatinamerica committed:
SPRINT 23 часть 4
Переход на новый фрагмент работает, но текста на нём нет. ВОобще ничего нет

CapitanLatinamerica committed:
SPRINT 23 часть 5
Поправил разметку фрагмента, немного исправил поведение при запуске. Нужно избавляться от меню навигации

CapitanLatinamerica committed:
SPRINT 23 часть 6
Скрыл меню нижней навигации, обработал "Назад" на тулбаре

CapitanLatinamerica committed:
SPRINT 23 часть 7
Исправил фрагмент - теперь отображается как надо! Но треки пока не загружены

CapitanLatinamerica committed:
SPRINT 23 часть 8
Испортил фрагмент. Пытаюсь передать треки в экран просмотра

CapitanLatinamerica committed:
SPRINT 23 часть 8.1
Вернул фрагмент в хороший вид, но треки не грузятся

CapitanLatinamerica committed:
SPRINT 23 часть 9
Загружаются треки из плейлиста. Я сделал так, что айтюнс сервис по айди их загружает и добавляет. Ну а почему бы и нет? Чтобы сыграть отрезок нам всё равно нужен интернет. Мы не храним в БД саму мелодию 😂😂

CapitanLatinamerica committed:
SPRINT 23 часть 10
Добавил метод удаления плейлиста.
РАБОТАЕТ ПО ДЛИННОМУ КЛИКУ НА ПЛЕЙЛИСТ!!!😂

CapitanLatinamerica committed:
SPRINT 23 часть 11
Добавил метод поделиться плейлистом

CapitanLatinamerica committed:
SPRINT 23 часть 12
Добавил вызов меню и вынес удаление плейлиста в Extension

CapitanLatinamerica committed:
SPRINT 23 часть 13
Поправил разметку BottomSheet для меню

CapitanLatinamerica committed:
SPRINT 23 часть 14
Сделал корректный вызов нижнего меню плейлиста

CapitanLatinamerica committed:
SPRINT 23 часть 15
Сделал переход по кнопке "Редактировать плейлист", но открывается пока только создание плейлиста.

CapitanLatinamerica committed:
SPRINT 23 часть 16
Всё открывается как надо. Осталось поправить количество минут

CapitanLatinamerica committed:
SPRINT 23 часть 17
Финалочка. Надеюсь это последний коммит проекта 😍

anastasianadelshtehel commented:
Действия на кнопки не выполняются, посмотри, можно нужно изменить верстку(может быть перекрытие другими элементами) или пересмотреть обработку кликов
https://github.com/user-attachments/assets/c8674565-eb5a-42b2-9c3e-87c289ea51fb

CapitanLatinamerica commented on app/src/main/java/com/practicum/playlistmaker/media/fragments/playlists/ui/view/MyAwesomePlaylistFragment.kt:
```diff
@@ -0,0 +1,324 @@
+        return inflater.inflate(R.layout.fragment_playlist_awesome, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
Ну что ж, значит надо сделать, если профессионал советует 😊

CapitanLatinamerica committed:
SPRINT 23 часть 18
Ну теперь-то точно всё! (но это не точно)
Так-с, что я сделал:

1) Удаление трека по долгому клику ✔ (я невнимательно читал задание, сделал удаление плейлиста по долгому клику сначала 😂 кстати, я это оставил сейчас - не ошибка?)
2) Вёрстка треков в плейлисте (ресайклер) - поправил (я надеюсь то поправил, что нужно было - отступ от левого края) ✔
3) Замечание: "Клик на кнопки поделиться и меню плохо срабатывает, даже при множественном нажатии может не отображаться действия". Тут я полностью согласен, у меня надо было прям точно по кнопке попасть, чтобы она сработала. Я так и знал, что будет замечание. Я сейчас расширил зону клика и должно теперь работать, но размер кнопок от дизайнера всё равно маловат на мой взгляд. ✔

4) Опционально: поправил некритичные замечания ✔
